src/tools/miri/src/intrinsics/mod.rs

```diff
@@ -308,6 +308,41 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                 this.write_scalar(res, dest)?;
             }
 
+            "fmuladdf32" => {
+                let [a, b, c] = check_arg_count(args)?;
+                let a = this.read_scalar(a)?.to_f32()?;
+                let b = this.read_scalar(b)?.to_f32()?;
+                let c = this.read_scalar(c)?.to_f32()?;
+                let fuse: bool = this.machine.rng.get_mut().gen();
+                let res = if fuse {
+                    // FIXME: Using host floats, to work around https://github.com/rust-lang/rustc_apfloat/issues/11
+                    let res = a.to_host().mul_add(b.to_host(), c.to_host()).to_soft();
+                    this.adjust_nan(res, &[a, b, c])
+                } else {
+                    let round = Round::NearestTiesToEven;
+                    let res = a.mul_r(b, round).value.add_r(c, round).value;
```

Comment:
`((a*b).value + c).value` should also work. Or `a.mul(b).value.add(c).value`, if you prefer that. But no reason to set the rounding mode explicitly.

Reply:
Oh, yes. Much nicer.